src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -293,8 +368,69 @@ private TextSpan GetRegionSpanForReference(SourceText sourceText, TextSpan refer
                     sourceText.Lines[lastLineNumber].End);
             }
 
+            private void UpdateCustomColumnsVisibility(MultiDictionary<string, string> customData)
+            {
+                // Check if we have any custom reference data to display.
+                if (customData.Count == 0)
+                {
+                    return;
+                }
+
+                // Get the new column states corresponding to the custom columns to display for custom data.
+                List<ColumnState2> newColumnStatesOpt = null;
```

Comment:
nit: it can be cleaner to just use ArrayBuilder, and simply free at the end.  but it's up to you.